Model/Feed.php

```diff
@@ -254,97 +280,107 @@ function sendCategories()
 
         $maxProgress = count($categoryCollection);
         $currentProgress = 0;
-        $isFirst = true;
+        $writtenCategories = false;
+        
+        if ($maxProgress > 0) {
+            $this->start(self::FEED_TYPE_CATEGORY);
```

Comment:
This should be done right before you write a category, so we can avoid all situations of writing an empty feed